tests/PHPStan/Analyser/data/trait-instance-of.php

```diff
@@ -0,0 +1,41 @@
+<?php declare(strict_types = 1);
+
+namespace TraitInstanceOf;
+
+use function PHPStan\Testing\assertType;
+
+trait Foo {
+	public function test(): string {
+		assertType('$this(TraitInstanceOf\HelloWorld)', $this);
+		if ($this instanceof HelloWorld) {
+			$this->bar();
+
+			assertType('$this(TraitInstanceOf\HelloWorld)', $this);
+			return 'hello world';
+		}
+		assertType('$this(TraitInstanceOf\HelloWorld~$this(TraitInstanceOf\HelloWorld))', $this);
+		if ($this instanceof OtherClass) {
+			$this->doOther();
+
+			assertType('$this(TraitInstanceOf\HelloWorld~$this(TraitInstanceOf\HelloWorld))&$this(TraitInstanceOf\OtherClass~$this(TraitInstanceOf\HelloWorld))', $this);
+			return 'other class';
+		}
+		assertType('$this(TraitInstanceOf\HelloWorld~$this(TraitInstanceOf\HelloWorld))', $this);
```

Comment:
I am not sure about this type here. maybe it could/should contain also that `OtherClass` is subtracted.
If so, I don't now what is missing to make this work